dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V7_0_2020_10_31__CollectionCommunity_Metadata_Handle.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.storage.rdbms.migration;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.dspace.handle.service.HandleService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.storage.rdbms.DatabaseUtils;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+
+/**
+ * Insert a 'dc.idendifier.uri' metadata record for each Community and Collection in the database.
+ * The value is calculated concatenating the canonicalPrefix extracted from the configuration
+ * (default is "http://hdl.handle.net/) and the object's handle suffix stored inside the handle table.
+ *
+ * @author Alessandro Martelli (alessandro.martelli at 4science.it)
+ */
+public class V7_0_2020_10_31__CollectionCommunity_Metadata_Handle extends BaseJavaMigration {
+    // Size of migration script run
+    protected Integer migration_file_size = -1;
+
+    @Override
+    public void migrate(Context context) throws Exception {
+
+        HandleService handleService = DSpaceServicesFactory
+                .getInstance().getServiceManager().getServicesByType(HandleService.class).get(0);
+
+        final String prefix = handleService.getCanonicalPrefix();
+
+        // Step 1 Preparation (creation of the tmp_handleprefix containing one record with the prefix)
+        String createTmpTable = "CREATE TABLE tmp_handleprefix (handleprefix text NOT NULL);";
+        DatabaseUtils.executeSql(context.getConnection(), createTmpTable);
+
+        // Step 2 Preparation (insertion of the prefix value into tmp_handleprefix)
+        String insertPrefixSql = "INSERT INTO tmp_handleprefix (handleprefix) VALUES (?)";
```

Comment:
@alemarte : Sorry, but I have to admit, I'm not a fan of this new logic to create a temp table to store the configured handle prefix.  I understand the reasoning behind it, but this approach seems more complex than necessary (and I'm also worried about fragility...if something errors out here, I'm worried some sites could end up with an unused `tmp_handleprefix` table sitting around).

Instead of trying to save the configuration into the database layer, it seems much easier to just use String variable substitution.  Afterall, you are reading the SQL migration script into a String (see `dataMigrateSQL` on line 63).   Therefore, why not do something like this:

```
// Get SQL script (based on DB type)
String dataMigrateSQL = MigrationUtils.getResourceAsString(sqlMigrationPath + "V7.0_2020.10.31__CollectionCommunity_Metadata_Handle.sql");

// Replace ${handle.canonical.prefix} variable in SQL script with value from Configuration
Map<String, String> valuesMap = new HashMap<>();
valuesMap.put("handle.canonical.prefix", handleService.getCanonicalPrefix());
StringSubstitutor sub = new StringSubstitutor(valuesMap);
dataMigrateSQL = sub.replace(dataMigrateSQL);

// Run the resulting script
DatabaseUtils.executeSql(context.getConnection(), dataMigrateSQL);
```

To me, this would be a lot cleaner & easier for other developers to understand.  It'd only require that you add a `${handle.canonical.prefix}` placeholder into the SQL scripts where it is necessary.  I'll also note that this *same idea* has already been used in [our FlywayUpgradeUtils class](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/storage/rdbms/FlywayUpgradeUtils.java#L97-L104), which uses  similar placeholder variables in the [`upgradeToFlyway4x.sql`](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/resources/org/dspace/storage/rdbms/flywayupgrade/postgres/upgradeToFlyway4x.sql) script.

Apologies that I didn't suggest this approach sooner, but I had forgotten it until I analyzed your new code.